frontend/src/app/tasks/week/page.tsx

```diff
@@ -1,3 +1,87 @@
+import { makeGetTasks } from '@/infra/task/getTasks'
+import { serverFetch } from '@/infra/lib/serverFetch'
+import { authOptions } from '@/app/api/auth/[...nextauth]/route'
+import { getServerSession } from 'next-auth'
+import { format, sub, eachDayOfInterval } from 'date-fns'
+import { Task } from '@/domain/Task'
+import { Typography, Box, Stack } from '@mui/joy'
+import { SimpleTaskBox } from '../components/TaskBox'
+import { convertTimeToMinutes, convertMinutesToTime } from '../utils/time'
+
+const getTasks = async (startTime: string, endTime: string) => {
+  const apiClient = await serverFetch()
+  const getTasks = makeGetTasks(apiClient)
+  const session = await getServerSession(authOptions)
+  const { id: userId } = session!.user
+
+  const tasks = await getTasks({ userId, startTime, endTime })
+
+  // Group tasks by date and add the total time for that date
+  return tasks.reduce<Record<Task['date'], { tasks: Array<Task>; time: number }>>((acc, task) => {
+    const startTime = convertTimeToMinutes(task.startTime)
+    const endTime = convertTimeToMinutes(task.endTime)
+    const timeDiff = endTime - startTime
+
+    if (acc[task.date]) {
+      const { tasks, time } = acc[task.date]
+
+      return { ...acc, [task.date]: { tasks: [...tasks, task], time: time + timeDiff } }
+    }
+
+    return { ...acc, [task.date]: { tasks: [task], time: timeDiff } }
+  }, {})
+}
+
 export default async function WeekView() {
-  return <div>Week!</div>
+  const today = new Date()
+  const sevenDaysAgo = sub(today, { days: 6 })
```

Comment:
I am pretty sure that we would want to display the current ISO week instead of today - 6 days. This would ensure that the grid here is always consistent - i.e., always Mon - Sun. @anarute what do you think?

Reply:
Yes, agree.

Reply:
Cool, I'm working on it!

Reply:
![image](https://github.com/Igalia/phpreport/assets/20492786/ee06f5aa-1793-439d-8aff-ac21ddb1665a)


Like this?